graylog2-web-interface/src/components/search/Decorator.jsx

```diff
@@ -63,22 +66,32 @@ const Decorator = React.createClass({
 
     return Object.assign({}, config, resolvedConfig);
   },
+  _formatActionsMenu() {
+    const permissions = this.state.currentUser.permissions;
+    const decorator = this.props.decorator;
+    const editPermission = this.isPermitted(permissions, `decorators:edit:${decorator.stream}`);
+    return (
+      <DropdownButton id={`decorator-${decorator._id}-actions`} bsStyle="default" bsSize="xsmall" title="Actions" pullRight>
+        <MenuItem onSelect={this._handleEditClick} disabled={!editPermission}>Edit</MenuItem>
+        <MenuItem divider/>
+        <MenuItem onSelect={this._handleDeleteClick} disabled={!editPermission}>Delete</MenuItem>
+      </DropdownButton>
+    );
+  },
   render() {
-    if (!this.state.types) {
+    if (!this.state.types || !this.state.currentUser) {
       return <Spinner />;
     }
     const decorator = this.props.decorator;
     const config = this._resolveConfigurationIds(decorator.config);
     const decoratorType = this.state.types[decorator.type] || this._decoratorTypeNotPresent();
+
+    const decoratorActionsMenu = this._formatActionsMenu();
     return (
       <span className={DecoratorStyles.fullWidth}>
         <div className={DecoratorStyles.decoratorBox}>
           <h6 className={DecoratorStyles.decoratorType}>{decoratorType.name}</h6>
-          <DropdownButton id={`decorator-${decorator._id}-actions`} bsStyle="default" bsSize="xsmall" title="Actions" pullRight>
-            <MenuItem onSelect={this._handleEditClick}>Edit</MenuItem>
-            <MenuItem divider/>
-            <MenuItem onSelect={this._handleDeleteClick}>Delete</MenuItem>
-          </DropdownButton>
+          {decoratorActionsMenu}
```

Comment:
Yeah, I agree with you on that one and your remark is totally correct. It's just a matter of balancing consistency and choosing the strategy which seems to make the most sense. I am also unsure which one's more important here. :)